markdown/stencil-docs/custom-javascript-implementations/adding-javascript.md

```diff
@@ -56,6 +58,133 @@ Many third-party JavaScript components are distributed with npm (Node Package Ma
+
+* `pages/account/addresses`
+* `pages/account/add-address`
+* `pages/account/add-return`
+* `pages/account/add-wishlist`
+* `pages/account/recent-items`
+* `pages/account/download-item`
+* `pages/account/edit`
+* `pages/account/return-saved`
+* `pages/account/returns`
+* `pages/auth/login`
+* `pages/auth/account-created`
+* `pages/auth/create-account`
+* `pages/auth/new-password`
+* `pages/blog`
+* `pages/blog-post`
+* `pages/brand`
+* `pages/brands`
+* `pages/cart`
+* `pages/category`
+* `pages/compare`
+* `pages/errors`
+* `pages/gift-certificate/purchase`
+* `pages/gift-certificate/balance`
+* `pages/gift-certificate/redeem`
+* `global`
+* `pages/home`
+* `pages/order-complete`
+* `pages/page`
+* `pages/product`
+* `pages/search`
+* `pages/sitemap`
+* `pages/subscribed`
+* `pages/account/wishlist-details`
+* `pages/account/wishlists`
+* `pages/order-confirmation`
+* `pages/checkout`
+
+These page types correspond to the pages within your theme. Each of these page types maps to an ES6 module that extends the base `PageManager` abstract class:
+
+<div class="HubBlock-header">
+    <div class="HubBlock-header-title flex items-center">
+        <div class="HubBlock-header-name">pages/blog extending the page PageManager class</div>
+    </div><div class="HubBlock-header-subtitle"></div>
+</div>
+
+<!--
+title: "pages/blog extending the page PageManager class"
+subtitle: ""
+lineNumbers: true
+-->
+
+```js
+export default class Blog extends PageManager {
+        constructor() {
+            //Setup code goes here – attach to internals, and use internals as you would 'this'
+        }
+    }
+```
+
+<div class="HubBlock--callout">
+<div class="CalloutBlock--">
+<div class="HubBlock-content">
+    
+<!-- theme:  -->
+
+> 1. Notice the page types correspond to the html pages in [cornerstone/templates/pages/](https://github.com/bigcommerce/cornerstone/tree/master/templates/pages)
+
+</div>
+</div>
+</div>
+
+---
+
+<a href='#adding_page-template-injection' aria-hidden='true' class='block-anchor'  id='page-types_javascript'><i aria-hidden='true' class='linkify icon'></i></a>
+
+## JavaScript Template Context Injection
+
+Occasionally, you might need to use dynamic data from the template context within your theme's client-side application code. Two helpers are provided to help achieve this.
+
+The `inject` helper allows you to compose a json object with a subset of the template context to be sent to the browser:
+
+`{{inject "stringBasedKey" contextValue}}`
+
+To retrieve the parsable JSON object, just call `{{jsContext}}` after all of the `{{inject}}` calls.
+
+For example, to set up the product name in your client-side app, you can do this if you're in the context of a product:
+
+<div class="HubBlock-header">
+    <div class="HubBlock-header-title flex items-center">
+        <div class="HubBlock-header-name"></div>
+    </div><div class="HubBlock-header-subtitle"></div>
+</div>
+
+<!--
+title: ""
+subtitle: ""
+lineNumbers: true
+-->

```

Comment:
Either add the labels, or remove the extra html and just leave it as a code block.